### PR TITLE
ykclient: backport fix for "make check"

### PR DESCRIPTION
**Pull request by mitchblank**
This formula was failing to bottle because some build-time checks failed.  I verified that these don't fail in HEAD however.

After some `git bisect` work, I found that this commit from @klali is needed to work against the current api.yubico.com service, so cherry-picking it until the next official release.

Not sure when an official release will happen -- 2.15 is 5 years old although the git version has seen at least sporadic development (most recent commit is a couple months ago)

### CODE REVIEW

**Comment by SMillerDev** [comment]
Made an issue https://github.com/Yubico/yubico-c-client/issues/52, but that shouldn't block this (just waiting for CI before I approve)

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**
